app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala

```diff
@@ -0,0 +1,149 @@
+      val propertyLinksConnector: PropertyLinkConnector,
+      businessRatesAttachmentService: BusinessRatesAttachmentsService,
+      val runModeConfiguration: Configuration,
+      occupancyOfPropertyView: views.html.propertyLinking.occupancyOfProperty,
+      serviceUnavailableView: views.html.errors.serviceUnavailable,
+      propertyLinkingService: PropertyLinkingService)(
+      implicit executionContext: ExecutionContext,
+      override val messagesApi: MessagesApi,
+      override val controllerComponents: MessagesControllerComponents,
+      val config: ApplicationConfig
+) extends PropertyLinkingController {
+
+  import ClaimPropertyOccupancy._
+
+  def showOccupancy(): Action[AnyContent] = authenticatedAction.andThen(withLinkingSession).async { implicit request =>
+    {
+
+      propertyLinkingService.findEarliestStartDate(request.ses.uarn).flatMap {
+        case Some(date) if date.isAfter(LocalDate.now()) =>
+          businessRatesAttachmentService
+            .persistSessionData(request.ses.copy(
+              propertyOccupancy = Some(PropertyOccupancy(stillOccupied = true, lastOccupiedDate = None))))
+            .map { _ =>
+              Redirect(routes.ChooseEvidenceController.show())
+            }
+            .recover {
+              case UpstreamErrorResponse.Upstream5xxResponse(_) =>
+                ServiceUnavailable(serviceUnavailableView())
+            }
+        case _ =>
+          val form = request.ses.propertyOccupancy.fold(occupancyForm()) { occupancy =>
+            occupancyForm(
+              startDate = request.ses.propertyOwnership.flatMap(_.fromDate),
+              errorMessageKeySuffix = request.ses.clientDetails.fold("")(_ => ".client")).fillAndValidate(occupancy)
+          }
+          Future.successful(
+            Ok(
+              occupancyOfPropertyView(
+                form,
+                request.ses.clientDetails,
+                controllers.propertyLinking.routes.ClaimPropertyOwnershipController.showOwnership().url)))
+      }
+    }
+
+  }
+
+  def submitOccupancy(): Action[AnyContent] =
+    authenticatedAction.andThen(withLinkingSession).async { implicit request =>
+      occupancyForm(
+        startDate = request.ses.propertyOwnership.flatMap(_.fromDate),
+        errorMessageKeySuffix = request.ses.clientDetails.fold("")(_ => ".client"))
+        .bindFromRequest()
+        .fold(
+          errors =>
+            Future.successful(
+              BadRequest(
+                occupancyOfPropertyView(
+                  errors,
+                  request.ses.clientDetails,
+                  controllers.propertyLinking.routes.ClaimPropertyOwnershipController.showOwnership().url))),
+          formData =>
+            businessRatesAttachmentService
+              .persistSessionData(request.ses.copy(propertyOccupancy = Some(formData)))
+              .map { _ =>
+                Redirect(routes.ChooseEvidenceController.show())
+              }
+              .recover {
```

Comment:
same here, i don't think we need to recover like this. let it fail and error handler will display service unavailable